esmvaltool/diag_scripts/portrait_plot.py

```diff
@@ -0,0 +1,554 @@
+        group_data = data.isel({cfg["group_by"]: i})
+        group_data = group_data.dropna(cfg["x_by"], how="all")
+        for sss in range(split_count):
+            split = group_data.isel({cfg["split_by"]: sss})
+            split_label = split.coords[cfg["split_by"]].values.item()
+            if split_label == cfg["default_split"]:
+                log.debug("Skipping default split for overlay.")
+                continue
+            nodes = get_triangle_nodes(sss, split_count)
+            overlay_reference(cfg, grid[i], split.values.T, nodes)
+
+
+def plot(cfg, data):
+    """Create figure with subplots for each group and save to NetCDF.
+
+    sets same color range and  overlays additional references based on
+    the content of data (xr.DataArray)
+    """
+    save_to_netcdf(cfg, data)
+
+    fig = plt.figure(1, cfg.get("figsize", (5.5, 3.5)))
+    group_count = len(data.coords[cfg["group_by"]])
+    grid = ImageGrid(
+        fig,
+        111,  # similar to subplot(111)
+        cbar_mode="single",
+        cbar_location="right",
+        cbar_pad=0.1,
+        cbar_size=0.2,
+        nrows_ncols=(1, group_count),
+        axes_pad=0.1,
+    )
+    # remap colorbar to 10 discrete steps
+    cmap = mpl.cm.get_cmap(cfg.get("cmap", "RdYlBu_r"), 10)
+    cfg["plot_kwargs"]["cmap"] = cmap
+    for i in range(group_count):
+        group = data.isel({cfg["group_by"]: i})
+        group = group.dropna(cfg["x_by"], how="all")
+        title = None
+        if group_count > 1:
+            title = group.coords[cfg["group_by"]].values.item()
+        plot_group(cfg, grid[i], group, title=title)
+    # use same colorrange and colorbar for all subplots:
+    unify_limits(grid)
+    # set cb of first image as single cb for the figure
+    grid.cbar_axes[0].colorbar(grid[0].get_images()[0], **cfg["cbar_kwargs"])
+    if data.shape[3] > 1:
+        plot_overlays(cfg, grid, data)
+    if cfg["plot_legend"] and data.shape[3] > 1:
+        split_legend(cfg, grid, data)
+    basename = "portrait_plot"
+    fname = get_plot_filename(basename, cfg)
+    plt.savefig(fname, bbox_inches="tight", dpi=cfg["dpi"])
+    with ProvenanceLogger(cfg) as prov_logger:
+        prov_logger.log(fname, get_provenance(cfg))
+    log.info("Figure saved:")
+    log.info(fname)
+
+
+def normalize(array, method, dims):
+    """Divide and shift values along dims depending on method."""
+    shift = 0
+    norm = 1
+    if "mean" in method:
+        norm = array.mean(dim=dims)
+    elif "median" in method:
+        norm = array.median(dim=dims)
+    if "centered" in method:
+        shift = norm
+    normalized = (array - shift) / norm
+    return normalized
+
+
+def apply_distance_metric(cfg, metas):
+    """Optionally apply preproc method.
+
+    reference_for_metric facet required.
+    """
+    if not cfg["distance_metric"]:
+        return
+    for y_metas in group_metadata(metas, cfg["y_by"]).values():
+        try:  # TODO: add select_single_metadata to shared?
```

Comment:
```suggestion
        try:
```
I guess obsolete since this doesn't exist in this file either?